addons/l10n_ec/models/account_move.py

```diff
@@ -0,0 +1,259 @@
+            document_allowed = self.env.ref('l10n_ec.%s' % document_ref, False)
+            if document_allowed:
+                documents_allowed |= document_allowed
+        return documents_allowed
+
+    def _get_l10n_latam_documents_domain(self):
+        if self.company_id.country_id.code != "EC":
+            return super(AccountMove, self)._get_l10n_latam_documents_domain()
+        internal_type = self.env.context.get("internal_type", False)
+        if internal_type == "purchase_liquidation":
+            internal_types = ["purchase_liquidation"]
+            domain = [
+                ("internal_type", "in", internal_types),
+                ("country_id", "=", self.company_id.country_id.id),
+            ]
+        else:
+            domain = super(AccountMove, self)._get_l10n_latam_documents_domain()
+        latam_type = internal_type or "invoice"
+        if self.move_type in ("out_refund", "in_refund"):
+            latam_type = "credit_note"
+        if self.debit_origin_id:
+            latam_type = "debit_note"
+        allowed_documents = self._get_l10n_ec_documents_allowed(self._get_l10n_ec_identification_type())
+        if latam_type and allowed_documents:
+            domain.append(
+                (
+                    "id",
+                    "in",
+                    allowed_documents.filtered(
+                        lambda x: x.internal_type == latam_type
+                    ).ids,
+                )
+            )
+        return domain
+
+    def _get_ec_formatted_sequence(self, number=0):
+        return "%s %s-%s-%09d" % (
+            self.l10n_latam_document_type_id.doc_code_prefix,
+            self.journal_id.l10n_ec_entity,
+            self.journal_id.l10n_ec_emission,
+            number,
+        )
+
+    def _get_starting_sequence(self):
+        """If use documents then will create a new starting sequence using the document type code prefix and the
+        journal document number with a 8 padding number"""
+        if (
+            self.journal_id.l10n_latam_use_documents
+            and self.env.company.country_id.code == "EC"
+        ):
+            if self.l10n_latam_document_type_id:
+                return self._get_ec_formatted_sequence()
+        return super()._get_starting_sequence()
+
+    def _get_last_sequence_domain(self, relaxed=False):
+        if self.country_code == "EC" and self.move_type not in (
+            "entry",
+            "in_receipt",
+            "out_receipt",
+        ):
+            relaxed = True
+            where_string, param = super(AccountMove, self)._get_last_sequence_domain(
+                relaxed
+            )
+            where_string += """
+            AND sequence_prefix like %(sequence_prefix)s
+            """
+            param["sequence_prefix"] = "%%%s-%s-" % (
+                self.journal_id.l10n_ec_entity,
+                self.journal_id.l10n_ec_emission,
+            )
+            if self.debit_origin_id:
```

Comment:
credit and debit notes always need to be created from the invoice and not manually.

Reply:
Analyze also this @cormaza https://github.com/odoo/odoo/blob/82788f6b6a2501412b2deff8da70867b67db4d6f/addons/l10n_ar/models/account_move.py#L229